src/reducers/workspaces.ts

```diff
@@ -359,6 +368,21 @@ export const reducer: Reducer<IWorkspaceManagerState> = (
           isRunning: false
         }
       };
+    case EVAL_TESTCASE_FAILURE:
+      return {
+        ...state,
+        [location]: {
+          ...state[location],
+          editorTestcases: state[location].editorTestcases.map((testcase, i) => {
+            if (i === index) {
```

Comment:
Why is `index` retrieved at the very top when your only using it for two cases? Why not get it from the `index` explicitly in the case itself so its clear where it is coming from? (its easier to see where it is coming from)

Reply:
Refactored to remove the constant `index`